internal/encoding/encoding.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Mia srl
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encoding
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	JSON = "json"
+	YAML = "yaml"
+)
+
+type MarshalOptions struct {
+	Indent bool
+}
+
+type UnsupportedEncodingError struct {
+	Encoding string
+}
+
+func (e UnsupportedEncodingError) Error() string {
+	return fmt.Sprintf("unsupported encoding: %s", e.Encoding)
+}
```

Comment:
I think it's a bit overkill to use a struct just to define an error, you could just:
```go
var ErrUnsupportedEncoding  = errrors.New("unsupported encoding")
```
and then  use it in the code: 
```go
fmt.Errorf("%w: %s", ErrUnsupportedEncoding, encoding)
```

Also I changed the error name to respect Go conventions

Reply:
@fredmaggiowski we had a discussion about this topic.
What do you think about?

Reply:
I agree with both of you, these are two different approaches that I'd use for different use cases; one is for building errors with custom data (as in this case), the other is for wrapping received errors with custom ones.

In this situation the approach taken by @pmarchini is fine for me